### PR TITLE
New package: Supernova.SupernovaPlayer version 0.92

### LINKED ISSUE #164152: [Update Request]: Supernova.SupernovaPlayer
### What type of update are you requesting?

A new version of an existing package

### Current Package Identifier

Supernova.SupernovaPlayer

### Package Version

0.88+

### Please describe the changes you would like to see

- Reference: https://github.com/microsoft/winget-pkgs/pull/164054

### PR DESCRIPTION
**Pull request by Dragon1573**
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- Resolve #164152

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/164367)

### CODE REVIEW

**Comment by Trenly** [comment]
@stephengillie - please do **not** manually merge this PR even if the feedback is addressed; Dragos will be looking into #142854 and having a reference PR open will be helpful

**Comment by Dragon1573** [comment]
This package should co-operate with browsers. When loading online `*.swf` resources, SuperNova Player will ask to launch.

File: manifests/s/Supernova/SupernovaPlayer/0.92/Supernova.SupernovaPlayer.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Created with komac v2.3.0
```
Comment (Trenly) [inline manifests/s/Supernova/SupernovaPlayer/0.92/Supernova.SupernovaPlayer.installer.yaml:1]:
```suggestion
# Created with komac v2.3.0 - new commit for testing
```
@Dragon1573 - the Policy Service team is asking for a new commit to be added so they can track down the cause of #142854; Can you commit this suggestion so they can continue their testing?

File: manifests/s/Supernova/SupernovaPlayer/0.92/Supernova.SupernovaPlayer.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Created with komac v2.3.0
```
Comment (Dragon1573) [inline manifests/s/Supernova/SupernovaPlayer/0.92/Supernova.SupernovaPlayer.locale.en-US.yaml:1]:
```suggestion
# Created with komac v2.3.0 - new commit for testing
```

File: manifests/s/Supernova/SupernovaPlayer/0.92/Supernova.SupernovaPlayer.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Created with komac v2.3.0
```
Comment (Dragon1573) [inline manifests/s/Supernova/SupernovaPlayer/0.92/Supernova.SupernovaPlayer.yaml:1]:
```suggestion
# Created with komac v2.3.0 - new commit for testing
```

**Comment by stephengillie** [comment]
> @ stephengillie - please do **not** manually merge this PR even if the feedback is addressed; Dragos will be looking into #142854 and having a reference PR open will be helpful

Adding https://github.com/microsoft/winget-pkgs/labels/Blocking-Issue label to prevent merging.

**Comment by stephengillie** [comment]
> This package should co-operate with browsers. When loading online `*.swf` resources, SuperNova Player will ask to launch.

If I'm understanding correctly, to validate this package, I'll need to open a webpage that displays a `.swf` file.

**Comment by Dragon1573** [comment]
> If I'm understanding correctly, to validate this package, I'll need to open a webpage that displays a `.swf` file.

Yep. It's designed for a replacement of _Adobe Flash Player_ and provide further support for Flash contents (especially `*.swf`) in any browser.

Currently, browser extension is only available at _Chrome Extension Store_. Can't sure if other browsers support it without installing an extension.

**Comment by Dragon1573** [comment]
> @stephengillie - please do **not** manually merge this PR even if the feedback is addressed; Dragos will be looking into #142854 and having a reference PR open will be helpful

@Trenly @stephengillie , could this pull request being merged now? 🙏🏼

**Comment by Trenly** [comment]
[Policy] Reset Labels

**Comment by stephengillie** [comment]
Hi @Dragon1573,

At first, the application installs to a `0.0.0` subfolder.
image
When the launcher application is first opened, it spends some time downlaoding updates. Is there a way to have the installer do this updating, so the application is immediately ready to use on first run? 

image
The OK box appears, and then the application opens a browser window, showing what looks like a "thank you" message that immediately closes. I couldn't pause the VM fast enough to get a good screenshot of the message, but here is the URL:
image

Now, in the `0.1.23` subfolder, there is a `player` folder, which has an application that doesn't display any UI:
image

Is this the expected behavior? Should I be testing this in another way?

Verify:
image
Since it doesn't write any version data to the Registry, and doesn't display any in any application that I could find, I can't verify the version number.

**Comment by Dragon1573** [comment]
Supernova team provides a `*.swf` game (Color Move) on their "Demo" page. Could you please try it?

It seems browser can't directly open a `*.swf` file using Supernova. The `*.swf` should be embedded into a web page.

- See https://github.com/microsoft/winget-pkgs/pull/164367#issuecomment-2250705709

**Comment by Dragon1573** [comment]
> When I try to load any `.swf` file, such as the `snplayer2.swf` included with the application, the browser doesn't seem to load it and instead just downloads it, giving a file type warning. I'm not sure how to verify this application.

> [!NOTE]
>
> 1. `*.swf` file can't be opened by Supernova directly. We have to use a special URL like `supernova://play?swfurl=https://cdn.example.com/path/to/file.swf` in the browser to help launching Supernova. You can simply try the provided URL below.
> 2. It seems Supernova does not support files with local file schemas (`file://`), you have to serve them to Supernova via a local web server with `http://` or `https://`.
>
> ```text
> supernova://play?swfurl=https://www.getsupernova.com/games/colormove.swf
> ```

**Comment by stephengillie** [comment]
Verify:
image
image
image
Their SSL cert expired 10 days ago.
image

**Review by stephengillie — APPROVED** [review]

**Pull request merged**